rxandroidble/src/main/java/com/polidea/rxandroidble/internal/connection/ConnectionComponent.java

```diff
@@ -7,17 +7,25 @@
 import dagger.Subcomponent;
 
 @ConnectionScope
-@Subcomponent(modules = ConnectionModule.class)
+@Subcomponent(modules = {ConnectionModule.class, ConnectionModuleBinder.class})
 public interface ConnectionComponent {
 
+    class NamedBooleans {
+        public static final String AUTO_CONNECT = "autoConnect";
+        public static final String SUPPRESS_OPERATION_CHECKS = "suppressOperationChecks";
+        private NamedBooleans() { }
+    }
+
     @Subcomponent.Builder
     interface Builder {
 
+        Builder connectionModule(ConnectionModule connectionModule);
+
         ConnectionComponent build();
     }
 
     @ConnectionScope
```

Comment:
I think further optimization is out of scope for this pull request. We should have a separate PR for optimizations if you think they are necessary.